src/devices/Mcp3428/Mcp3428Async.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Device.I2c;
+using System.IO;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Iot.Device.Mcp3428
+{
+    using static Helpers;
+    /// <summary>
+    /// Async API is mostly useful for greater resolutions and one-shot mode, when conversion time can be significant.
+    /// Implements the <see cref="Iot.Device.Mcp3428.Mcp3428" />
+    /// </summary>
+    /// <seealso cref="Iot.Device.Mcp3428.Mcp3428" />
+    /// <autogeneratedoc />
+    public class Mcp3428Async : Mcp3428
```

Comment:
Any particular reason why we chose to separate Async code into a different class? I would rather follow the .NET patterns and have them all in the same class, and simply naming the methods that are Async with the `Async` suffix.

Reply:
I wasn't sure in the async interface for a long time. Decided to wait for review and fix later.   
The other reason was that I have some caching and background update ideas for my use case, where I have to poll multiple devices. It seamed to be reasonable to cut the API to a "sync and just works" and a more feature-rich implementation. This never made it into the PR...  
Fixing it right now.